File: 0560-subarray-sum-equals-k/0560-subarray-sum-equals-k.py
class Solution:
    def subarraySum(self, nums: List[int], k: int) -> int:
        # 이전에 나온 합 개수
        sum_count = {0: 1}
        answer = 0
        s = 0
        for num in nums:
            s += num
            answer += sum_count.get(s - k, 0)
            sum_count[s] = sum_count.get(s, 0) + 1
        
        return answer

File: 0560-subarray-sum-equals-k/0560-subarray-sum-equals-k.cpp
class Solution {
public:
    int subarraySum(vector<int>& nums, int k) {
        map<int, int> sum_count;
        sum_count[0] = 1;
        int answer = 0, sum = 0;
        for (const int& num : nums) {
            sum += num;
            answer += sum_count[sum - k];
            sum_count[sum] += 1;
        }
        return answer;
    }
};